src/app/shared/components/course-form/course-form.component.ts

```diff
@@ -1,20 +1,109 @@
 import { Component } from '@angular/core';
-import { FormBuilder, FormGroup } from '@angular/forms';
+import { FormArray, FormBuilder, FormGroup, Validators } from '@angular/forms';
 import { FaIconLibrary } from '@fortawesome/angular-fontawesome';
 import { fas } from '@fortawesome/free-solid-svg-icons';
+import { ButtonText, IconNames } from '@shared/models/button.model';
+import { mockedAuthorsList } from '@shared/mocks/mocks';
+import { Author } from '@shared/models/author.model';
 
 @Component({
   selector: 'app-course-form',
   templateUrl: './course-form.component.html',
   styleUrls: ['./course-form.component.scss'],
 })
 export class CourseFormComponent {
+  courseForm!: FormGroup;
+  allAuthors: Author[] = [...mockedAuthorsList];
+  courseAuthors: Author[] = [];
+  protected readonly ButtonText = ButtonText;
+  protected readonly IconNames = IconNames;
+
   constructor(
     public fb: FormBuilder,
     public library: FaIconLibrary
   ) {
     library.addIconPacks(fas);
+    this.buildForm();
+  }
+
+  buildForm() {
+    this.courseForm = this.fb.group({
+      title: ['', [Validators.required, Validators.minLength(2)]],
+      description: ['', [Validators.required, Validators.minLength(2)]],
+      courseAuthors: this.fb.array([]),
+      newAuthor: this.fb.group({
+        author: [
+          '',
+          [Validators.minLength(2), Validators.pattern('^[a-zA-Z0-9]*$')],
+        ],
+      }),
+      duration: [0, [Validators.required, Validators.min(0)]],
+    });
+  }
+
+  addAuthorToCourse(author: Author) {
+    this.allAuthors = this.allAuthors.filter(athr => athr.id !== author.id);
+
+    this.courseAuthors.push(author);
+    this.courseAuthorsArray.push(this.fb.control(author));
+  }
+
+  removeAuthorFromCourse(author: Author) {
+    this.courseAuthors = this.courseAuthors.filter(
+      athr => athr.id !== author.id
+    );
+
+    this.allAuthors.push(author);
+
+    const index = this.courseAuthorsArray.controls.findIndex(
+      c => c.value.id === author.id
+    );
+    if (index >= 0) {
+      this.courseAuthorsArray.removeAt(index);
+    }
+  }
+
+  createNewAuthor() {
+    const authorName = this.newAuthorGroup.get('author')?.value;
+
+    if (authorName && authorName.length >= 2) {
+      const newAuthor = {
+        id: Math.floor(Math.random() * 1000).toString(),
```

Comment:
Im not totally sure, but as per task requirements, this should be a uuid (utilize a npm package).